controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/Endpoint.java

```diff
@@ -219,30 +226,41 @@ public static EndpointBuilder of(ApplicationId application) {
         private Port port;
         private boolean legacy = false;
         private boolean directRouting = false;
+        private boolean wildcard = false;
 
         private EndpointBuilder(ApplicationId application) {
             this.application = application;
         }
 
-        /** Sets the cluster and zone target of this  */
-        public EndpointBuilder target(ClusterSpec.Id cluster, ZoneId zone) {
```

Comment:
The idea here is that you can't have one without the the other.

Consider keeping `target(ClusterSpec.Id, ZoneId)` as-is and have both `wildcard()` and `wildcard(ZoneId)`. Should result in fewer validation rules I think.